core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -161,33 +148,22 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
         aw.persistenceId,
         aw.payload.map { pr =>
           val (pr2, tags) = pr.payload match {
-            case Tagged(payload, tags) if config.enableEventsByTagQuery =>
-              (pr.withPayload(payload), tags)
-            case _ => (pr, Set.empty[String])
+            case Tagged(payload, ts) =>
+              (pr.withPayload(payload), ts)
+            case _ =>
+              (pr, Set.empty[String])
           }
           serializeEvent(pr2, tags)
         }
       )
 
-    def publishTagNotification(serialized: Seq[SerializedAtomicWrite], result: Future[_]): Unit = {
-      if (pubsub.isDefined) {
-        result.foreach { _ =>
-          for (
-            p <- pubsub;
-            tag: String <- serialized.map(_.payload.map(_.tags).flatten).flatten.toSet
-          ) {
-            p ! DistributedPubSubMediator.Publish("akka.persistence.cassandra.journal.tag", tag)
-          }
-        }
-      }
-    }
-
     val p = Promise[Done]
     val pid = messages.head.persistenceId
     writeInProgress.put(pid, p.future)
 
-    Future(messages.map(serialize)).flatMap { serialized =>
-      val result =
+    // Is there a reason we dispatch the serialisation?
+    Future.successful(messages.map(serialize)).flatMap { serialized: Seq[SerializedAtomicWrite] =>
```

Comment:
The reason was that serialization is typically rather costly and by dispatching that more than one can be done at a time, otherwise it's done by the journal actor.

Reply:
Gotcha, reverted this